main/src/library/Order.flix

```diff
@@ -173,13 +183,18 @@ instance Order[Float32] {
 
     pub override def greaterEqual(x: Float32, y: Float32): Bool = $FLOAT32_GE$(x, y)
 
-    pub def compare(x: Float32, y: Float32): Comparison = match (x <= y, y <= x) {
-        case (true, true) => EqualTo
-        case (false, false) => EqualTo
-        case (true, false) => LessThan
-        case (false, true) => GreaterThan
-        case _ => EqualTo // TODO remove this case when PatternExhaustiveness is fixed
-    }
+    pub def compare(x: Float32, y: Float32): Comparison =
+        if (x <= y) {
```

Comment:
They say that List programmers knows the value of everything, but the cost of nothing.

- `x <= y ` is syntax for `Order.lessEqual` (i.e. a ***polymorphic call*** which you have in ***non-tail*** position).

(To be fair- how could you know :))

Does Flix optimize such calls? Does the JVM optimize such calls? Who knows!

ALSO,

I see 4 branches with TWO calls to `<=`. But I CAN DO BETTER 🥇 

Try:

```
if ($FLOAT32_LT$(x, y))
  LessThan 
else (
  if ($FLOAT32_GT$(x, y) GreaterThan else Equal
)
```

Note how half the time I save half the work!

Reply:
Good point, the original code was even more inefficient than I realized at first :)